src/app/content/hooks/locationChange/resolveContent.ts

```diff
@@ -148,50 +144,31 @@ const resolvePage = async(
   }
 };
 
+const getInputReferenceInfo = (bookId: string, inputVersion?: string) => {
+  const defaultVersion = BOOKS[bookId] ? BOOKS[bookId].defaultVersion : undefined;
+  const bookVersion = inputVersion ? inputVersion : defaultVersion;
+  return {bookId, bookVersion};
+};
+
 export const getBookInformation = async(
   services: AppServices & MiddlewareAPI,
   reference: ReturnType<typeof getContentPageReferences>[number]
 ) => {
-  const getInputReferenceInfo = (id: string, inputVersion?: string) => {
-    const defaultVersion = BOOKS[id] ? BOOKS[id].defaultVersion : undefined;
-    const bookVersion = inputVersion ? inputVersion : defaultVersion;
-
-    if (!bookVersion) {
-      return [];
-    }
-
-    return [{id, bookVersion}];
-  };
-
-  const allReferences = reference.bookId ? getInputReferenceInfo(reference.bookId, reference.bookVersion)
-    : await services.archiveLoader.getBookIdsForPage(reference.pageId);
-
-  const configuredReference = allReferences.filter((item) => BOOKS[item.id])[0];
+  const { bookId, bookVersion } = getInputReferenceInfo(reference.bookId, reference.bookVersion);
 
-  if (configuredReference) {
-    const osWebBook =  await services.osWebLoader.getBookFromId(configuredReference.id);
-    const archiveBook = await services.archiveLoader.book(
-      configuredReference.id, BOOKS[configuredReference.id].defaultVersion
-    ).load().catch((error) => {
+  const osWebBook =  await services.osWebLoader.getBookFromId(bookId);
+  const archiveBook = await services.archiveLoader.book(bookId, bookVersion || 'undefined')
```

Comment:
passing "undefined" here will always fail, you could probably short circuit earlier